src/config/config_impl.go

```diff
@@ -313,9 +313,14 @@ func (this *rateLimitConfigImpl) GetLimit(
 		logger.Debugf("looking up key: %s", finalKey)
 		nextDescriptor := descriptorsMap[finalKey]
 
-		if nextDescriptor == nil && len(prevDescriptor.wildcardKeys) > 0 {
-			for _, wildcardKey := range prevDescriptor.wildcardKeys {
-				if strings.HasPrefix(finalKey, strings.TrimSuffix(wildcardKey, "*")) {
+		if nextDescriptor == nil && len(prevDescriptor.wildcardValues) > 0 {
+			for _, wildcardValue := range prevDescriptor.wildcardValues {
+				wildcardKey := entry.Key + "_" + wildcardValue
+				if strings.HasSuffix(entry.Value, strings.TrimPrefix(wildcardValue, "*")) {
+					nextDescriptor = descriptorsMap[wildcardKey]
+					break
+				}
+				if strings.HasPrefix(entry.Value, strings.TrimSuffix(wildcardValue, "*")) {
```

Comment:
Are there any performance implications here due to the string comparison?

Reply:
I didn't check that for two reasons:
- I don't know how to perform a meaningful benchmark with regards to this piece of code
- Current code is already doing a string comparison in the same way